cmd/cluster/resize/infra_node.go

```diff
@@ -0,0 +1,308 @@
+			return false, err
+		}
+
+		readyNodes := 0
+		log.Printf("waiting for %d infra nodes to be reporting Ready", int(*originalMp.Spec.Replicas)*2)
+		for _, node := range nodes.Items {
+			for _, cond := range node.Status.Conditions {
+				if cond.Type == corev1.NodeReady {
+					if cond.Status == corev1.ConditionTrue {
+						readyNodes++
+						log.Printf("found node %s reporting Ready", node.Name)
+					}
+				}
+			}
+		}
+
+		switch {
+		case readyNodes >= int(*originalMp.Spec.Replicas)*2:
+			return true, nil
+		default:
+			log.Printf("found %d infra nodes reporting Ready, continuing to wait", readyNodes)
+			return false, nil
+		}
+	}); err != nil {
+		return err
+	}
+
+	// Delete temp machinepool
+	log.Printf("deleting temporary machinepool %s, with machine type %s", tempMp.Name, tempMp.Spec.Platform.AWS.InstanceType)
+	if err := r.hiveAdmin.Delete(ctx, tempMp); err != nil {
+		return err
+	}
+
+	// Wait for temporary machinepool to delete
+	if err := wait.PollImmediate(twentySecondIncrement, twentyMinuteTimeout, func() (bool, error) {
+		mp := &hivev1.MachinePool{}
+		err := r.hive.Get(ctx, client.ObjectKey{Namespace: tempMp.Namespace, Name: tempMp.Name}, mp)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return true, nil
+			}
+			return false, err
+		}
+
+		log.Printf("temporary machinepool %s/%s still exists, continuing to wait", tempMp.Namespace, tempMp.Name)
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	// Wait for infra node count to return to normal
+	log.Printf("waiting for infra node count to return to: %d", int(*originalMp.Spec.Replicas))
+	if err := wait.PollImmediate(twentySecondIncrement, twentyMinuteTimeout, func() (bool, error) {
+		nodes := &corev1.NodeList{}
+		selector, err := labels.Parse("node-role.kubernetes.io/infra=")
+		if err != nil {
+			return false, err
+		}
+
+		if err := r.client.List(ctx, nodes, &client.ListOptions{LabelSelector: selector}); err != nil {
+			return false, err
+		}
+
+		switch len(nodes.Items) {
+		case int(*originalMp.Spec.Replicas):
+			log.Printf("found %d infra nodes, infra resize complete", len(nodes.Items))
+			return true, nil
+		default:
+			log.Printf("found %d infra nodes, continuing to wait", len(nodes.Items))
+			return false, nil
+		}
+	}); err != nil {
+		return err
+	}
+
+	log.Printf("[REMINDER] follow the cleanup tasks in https://github.com/openshift/ops-sop/blob/master/v4/howto/resize-infras-workers.md#sending-the-all-clear")
+	return nil
+}
+
+func (r *Resize) getInfraMachinePool(ctx context.Context) (*hivev1.MachinePool, error) {
+	ns := &corev1.NamespaceList{}
+	selector, err := labels.Parse(fmt.Sprintf("api.openshift.com/id=%s", r.clusterId))
+	if err != nil {
+		return nil, err
+	}
+
+	if err := r.hive.List(ctx, ns, &client.ListOptions{LabelSelector: selector, Limit: 1}); err != nil {
+		return nil, err
+	}
+	if len(ns.Items) == 0 {
+		return nil, fmt.Errorf("found 0 namespaces with tag: api.openshift.com/id=%s", r.clusterId)
+	}
```

Comment:
**Question:** Should we also check that there's not more than 1 namespace matching here?

Reply:
Yeah, that's true, I should error when the list of matching namespaces isn't exactly 1